### ISSUE TITLE
Can't build from source

### ISSUE THREAD

**Issue by fawwaz** [line/line-bot-sdk-nodejs#11]
Dear Line official, 

I want to modify your bot client to extend some functionality. Since then, I tried to build your sdk from source by running these commands : 

```
git clone https://github.com/line/line-bot-sdk-nodejs
cd line-bot-sdk-nodejs
npm install
npm run build
```

However, I got some error message like this : 
```
> @line/bot-sdk@2.0.0 prebuild /home/fawwaz.muhammad/personal/line-bot-sdk-nodejs
> npm run lint && npm run clean


> @line/bot-sdk@2.0.0 lint /home/fawwaz.muhammad/personal/line-bot-sdk-nodejs
> tslint '{lib,test}/**/*.ts'


ERROR: lib/client.ts[19, 7]: The key 'messages' is not sorted alphabetically
ERROR: lib/client.ts[26, 7]: The key 'messages' is not sorted alphabetically
ERROR: lib/client.ts[33, 7]: The key 'messages' is not sorted alphabetically


npm ERR! Linux 4.4.0-79-generic
npm ERR! argv "/usr/bin/nodejs" "/home/fawwaz.muhammad/personal/line-bot-sdk-nodejs/node_modules/.bin/npm" "run" "lint"
npm ERR! node v4.8.2
npm ERR! npm  v3.7.5
npm ERR! code ELIFECYCLE
npm ERR! @line/bot-sdk@2.0.0 lint: `tslint '{lib,test}/**/*.ts'`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the @line/bot-sdk@2.0.0 lint script 'tslint '{lib,test}/**/*.ts''.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the @line/bot-sdk package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     tslint '{lib,test}/**/*.ts'
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs @line/bot-sdk
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls @line/bot-sdk
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/fawwaz.muhammad/personal/line-bot-sdk-nodejs/npm-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @line/bot-sdk@2.0.0 prebuild: `npm run lint && npm run clean`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @line/bot-sdk@2.0.0 prebuild script 'npm run lint && npm run clean'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the @line/bot-sdk package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run lint && npm run clean
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs @line/bot-sdk
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls @line/bot-sdk
npm ERR! There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/fawwaz.muhammad/.npm/_logs/2017-06-15T16_08_55_129Z-debug.log
```
I checked my node version and it satisfy your requirement 
```
node --version
v 4.8.2
```
What should I do ?

**Comment by fawwaz** [comment]
Well, the problem solved by modifying the `client.ts` code such the key returns in alphabetical order 
e.g like this : 
```
  public pushMessage(to: string, messages: Line.Message | Line.Message[]): Promise<any> {
    return this.post(URL.push, {
      messages: toArray(messages),
      to,
    });
  }

  public replyMessage(replyToken: string, messages: Line.Message | Line.Message[]): Promise<any> {
    return this.post(URL.reply, {
      messages: toArray(messages),
      replyToken,
    });
  }

  public multicast(to: string[], messages: Line.Message | Line.Message[]): Promise<any> {
    return this.post(URL.multicast, {
      messages: toArray(messages),
      to,
    });
  }
```

However I got another error like this : 
```
> @line/bot-sdk@2.0.0 prebuild /home/fawwaz.muhammad/personal/line-bot-sdk-nodejs
> npm run lint && npm run clean


> @line/bot-sdk@2.0.0 lint /home/fawwaz.muhammad/personal/line-bot-sdk-nodejs
> tslint '{lib,test}/**/*.ts'




> @line/bot-sdk@2.0.0 clean /home/fawwaz.muhammad/personal/line-bot-sdk-nodejs
> rm -rf dist


> @line/bot-sdk@2.0.0 build /home/fawwaz.muhammad/personal/line-bot-sdk-nodejs
> tsc

lib/middleware.ts(30,44): error TS2345: Argument of type 'string | string[]' is not assignable to parameter of type 'string'.
  Type 'string[]' is not assignable to type 'string'.
lib/middleware.ts(31,75): error TS2345: Argument of type 'string | string[]' is not assignable to parameter of type 'string'.
  Type 'string[]' is not assignable to type 'string'.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @line/bot-sdk@2.0.0 build: `tsc`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the @line/bot-sdk@2.0.0 build script 'tsc'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the @line/bot-sdk package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     tsc
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs @line/bot-sdk
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls @line/bot-sdk
npm ERR! There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/fawwaz.muhammad/.npm/_logs/2017-06-15T16_18_16_492Z-debug.log
```

**Comment by noraesae** [comment]
Thank you for your report. I confirmed the bug with the latest compiler and tools. I will upload commits to fix the problem soon.

**Comment by noraesae** [comment]
It's fixed in 972c2838c3a4a593381c2c7a33799ccb8db9bb60 and 9687a3eefa4156e8287781be60cc07d733471ebc. Thanks again!
